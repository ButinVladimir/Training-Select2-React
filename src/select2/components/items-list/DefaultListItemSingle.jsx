import React from 'react';
import PropTypes from 'prop-types';

export default function DefaultListItemSingle({
  value,
  name,
  checked,
  onChange,
}) {
  return (
    <li className={checked ? 'checked' : ''}>
      <button type="button" onClick={() => onChange(value, name)}>
        {name}
      </button>
    </li>
  );
}

DefaultListItemSingle.propTypes = {
  value: PropTypes.any.isRequired,
  name: PropTypes.string.isRequired,
  checked: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
};
