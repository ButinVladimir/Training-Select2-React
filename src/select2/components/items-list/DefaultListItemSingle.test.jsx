import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import DefaultListItemSingle from './DefaultListItemSingle';

describe('DefaultListItemSingle', () => {
  beforeAll(() => {
    Enzyme.configure({ adapter: new Adapter() });
  });

  test('renders without crashing unchecked', () => {
    const mockFn = jest.fn();
    const component = renderer.create(<DefaultListItemSingle
      checked={false}
      name="name"
      value="value"
      onChange={mockFn}
    />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(mockFn).not.toHaveBeenCalled();
  });

  test('renders without crashing checked', () => {
    const mockFn = jest.fn();
    const component = renderer.create(<DefaultListItemSingle
      checked
      name="name"
      value="value"
      onChange={mockFn}
    />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(mockFn).not.toHaveBeenCalled();
  });

  test('fires onChange properly', () => {
    const mockFn = jest.fn();
    const wrapper = mount(<DefaultListItemSingle
      checked={false}
      name="name"
      value="value"
      onChange={mockFn}
    />);

    const button = wrapper.find('button');
    button.simulate('click', 'value', 'name');
    expect(mockFn).toHaveBeenCalledWith('value', 'name');
  });
});
