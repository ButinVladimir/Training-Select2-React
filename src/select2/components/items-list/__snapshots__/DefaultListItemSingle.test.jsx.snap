// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultListItemSingle renders without crashing checked 1`] = `
<li
  className="checked"
>
  <button
    onClick={[Function]}
    type="button"
  >
    name
  </button>
</li>
`;

exports[`DefaultListItemSingle renders without crashing unchecked 1`] = `
<li
  className=""
>
  <button
    onClick={[Function]}
    type="button"
  >
    name
  </button>
</li>
`;
